app/views/cloud_volume/_common_new_edit.html.haml

```diff
@@ -62,24 +62,37 @@
     %span.help-block{"ng-show" => "angularForm.name.$error.required"}
       = _("Required")
 
-.form-group{"ng-class" => "{'has-error': angularForm.aws_volume_type.$invalid}",
+.form-group{"ng-class" => "{'has-error': angularForm.volume_type.$invalid}",
             "ng-if"    => "vm.cloudVolumeModel.emstype == 'ManageIQ::Providers::Amazon::StorageManager::Ebs'"}
   %label.col-md-2.control-label
     = _('Cloud Volume Type')
   .col-md-8
-    %select{"name"        => "aws_volume_type",
-            "ng-model"    => "vm.cloudVolumeModel.aws_volume_type",
-            "ng-options"  => "voltype.type as voltype.name for voltype in vm.awsVolumeTypes",
-            "ng-change"   => "vm.awsVolumeTypeChanged(vm.cloudVolumeModel.aws_volume_type)",
-            "ng-disabled" => "!vm.newRecord && vm.cloudVolumeModel.aws_volume_type == 'standard'",
+    %select{"name"        => "volume_type",
+            "ng-model"    => "vm.cloudVolumeModel.volume_type",
+            "ng-options"  => "voltype.type as voltype.name for voltype in vm.volumeTypes",
+            "ng-change"   => "vm.awsVolumeTypeChanged(vm.cloudVolumeModel.volume_type)",
+            "ng-disabled" => "!vm.newRecord && vm.cloudVolumeModel.volume_type == 'standard'",
             "required"    => "",
             :checkchange  => true,
             "miq-select"  => true}
       %option{"value" => "", "disabled" => ""}
         = "<#{_('Choose')}>"
-    %span.help-block{"ng-show" => "angularForm.aws_volume_type.$error.required"}
+    %span.help-block{"ng-show" => "angularForm.volume_type.$error.required"}
       = _("Required")
 
+.form-group{"ng-class" => "{'has-error': angularForm.volume_type.$invalid}",
+            "ng-if"    => "vm.cloudVolumeModel.emstype == 'ManageIQ::Providers::Openstack::StorageManager::CinderManager'"}
```

Comment:
We want to avoid constructs like this as much as possible (i.e. stating specific model name in UI code).

Ideally, this test should happen on the ruby side, using our `supports?` functionality and just propagate a boolean of some sort to the angular code & the view.

Reply:
I'm not aware of `supports?` being accessible through the API, and the user can change the Storage Manager form field on the fly, so I can't pre-inject a boolean value into the form. With that in mind, what is the appropriate UI pattern for dynamically accessing the result of `supports_whatever_feature?` when the Storage Manager selection is changed?

Reply:
I guess we don't have a universal pattern for this really. But I guess results of the supports? call can come with the storage managers.

Reply:
Nevermind, apparently there's already precedent for doing exactly that. Still seems incredibly messy, but I suppose that's how it has to be done.